arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -410,29 +411,33 @@ def checkFootnote(elt: ModelInlineFootnote, text: str) -> None:
                 ixExcludeTag = ixNStag + "exclude"
                 ixTupleTag = ixNStag + "tuple"
                 ixFractionTag = ixNStag + "fraction"
-                for elt, depth in etreeIterWithDepth(ixdsHtmlRootElt):
-                    eltTag = elt.tag
-                    if isinstance(elt, (_ElementTree, _Comment, _ProcessingInstruction, EntityBase)):
+                # Elsewhere in this module, we can use isinstance(elt, _Element) for typing purposes
+                # If we do that here, tests fail. Hence, for now, we define elt as _elt and use Any.
+                _elt: Any
+                for _elt, depth in etreeIterWithDepth(ixdsHtmlRootElt):
```

Comment:
Having two very similar variable names is error prone and likely to introduce bugs (you missed renaming some of the instances of `elt` below).

Revert the variable renames and cast here instead:
```python
for uncast_elt, depth in etreeIterWithDepth(ixdsHtmlRootElt):
    # Elsewhere in this module, we can use isinstance(elt, _Element) for typing purposes
    # If we do that here, tests fail. For now we cast elt as Any.
    elt = cast(Any, uncast_elt)
```

Reply:
Nice!